src/test/java/gov/cancer/tests/cts/advanced_search_form/CancerTypeSection_Test.java

```diff
@@ -0,0 +1,606 @@
+   */
+  @Test(dataProvider = "getInvalidSideEffect")
+  public void verifyInvalidSideEffectEntries(String path, String cancerType,String sideEffect, String error) {
+    TestRunner.run(AdvancedSearchPage.class, path, (AdvancedSearchPage page) -> {
+      page.getCancerTypeSection().scrollUntilAllButtonVisible();
+      //Click All button
+      page.getCancerTypeSection().getAllButton().click();
+      //selecting the Cancer Type
+      page.getCancerTypeSection().getCancerTypeCondition().selectItem(cancerType);
+      //selecting the Sub Type
+      page.getCancerTypeSection().getSideEffects().enterText(sideEffect);
+      //assert that message displayed is correct
+      Assert.assertEquals(page.getCancerTypeSection().getSideEffects().getSectionFieldMessage(),
+        error, "Correct message is not displayed for No available options");
+    });
+  }
+
+  /************************
+   * SUBMIT SEARCH WITH HITTING ENTER KEY
+   ***********************/
+  /** Verify that when user inputs a value to the Primary Cancer Type/Condition,Stage and Side Effects Fields and
+   * hit ENTER key, correct parameters are passed.
+   * parameters
+   * @param path       url
+   * @param cancerType cancer type value
+   * @param stage stage value
+   * @param sideEffect side effect value
+   * @param queryParam list of query parameters
+   */
+  @Test(dataProvider = "getCancerTypeConditionStageSideEffects")
+  public void verifyPrimaryCancerTypeStageSideEffectsFieldOnEnter(String path, String cancerType,String stage,
+                                                           String sideEffect, String queryParam) {
+    TestRunner.run(AdvancedSearchPage.class, path, (AdvancedSearchPage page) -> {
+      page.getCancerTypeSection().scrollUntilAllButtonVisible();
+      //Click All button
+      page.getCancerTypeSection().getAllButton().click();
+      //selecting the Cancer Type
+      page.getCancerTypeSection().getCancerTypeCondition().selectItem(cancerType);
+      //selecting the Stage
+      page.getCancerTypeSection().getStage().selectItem(stage);
+      //selecting the Side Effect
+      page.getCancerTypeSection().getSideEffects().selectItem(sideEffect);
+
+      // create a navigation event redirect to search results page
+      SearchNavigationResult search_result = page.hitEnterKey(page.getCancerTypeSection().getSideEffects());
+      //create a comparator object from the redirected URL query part
+      QueryParametersComparator querycomp = new QueryParametersComparator(search_result.getPageURL().getQuery());
+      //Assert that url contains expected query
+      Assert.assertTrue(querycomp.compareQueryParams(queryParam),
+        " query param does not match for Cancer Type, Stage and Side effects");
+    });
+  }
+  /************************************************************
+   * Data providers
+   ************************************************************/
+
+  /**
+   * Data provider retrieves paths to Advanced Search page
+   * params
```

Comment:
Data Providers comment should be more specific

Reply:
updated the comments